class.jetpack-gutenberg.php

```diff
@@ -146,7 +145,16 @@ static function add_block( $type, $args, $availability ) {
 		self::register( $type, $args, $availability );
 	}
 
-	static function load( $request = null ) {
+	static function load( $response = null, $handler = null, $request = null ) {
+		// We display beta blocks in available Gutenberg extensions endpoint requests
+		$is_availability_endpoint_beta = ! is_null( $request ) && $request->get_param( 'beta' ) && wp_endswith( $request->get_route(), 'gutenberg/available-extensions' );
+		// We display beta blocks in proxied requests
+		$is_proxied = function_exists( 'wpcom_is_proxied_request' ) ? wpcom_is_proxied_request() : false;
```

Comment:
I am being annoying 😓, what do you think about going with a filter instead?

Reply:
I don't see how a filter would be more useful here - could you please clarify, or provide an example?

Reply:
Doing something like `https://github.com/Automattic/jetpack/pull/11015/files#r243281349`, this way we don't need to do the `function_exists` check in Jetpack and expose 'wpcom_is_proxied_request`.

Reply:
Done in https://github.com/Automattic/jetpack/pull/11015/commits/85402077c575adea172ce03030bcfd950b9eda67

Reply:
And I've taken advantage of the filter in D22588-code.